src/components/Search.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+class Search extends React.Component {
+  constructor(){
+    super();
+
+    this.state = {
+      searchQuery: ""
+    }
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.receivePaginationQuery = this.receivePaginationQuery.bind(this);
+
+  }
+
+handleChange(event){
+  this.setState({
+    searchQuery: event.target.value
+  })
+  let searchEntry = event.target.value
+  if (searchEntry.length > 2){
+    handleSubmit(event)
```

Comment:
you are passing a change event into a submit event handler which might make things go funny. It would be better to split out functionality you want to share between two event handlers into own function and call that from both event handlers